docs/architecture/adr-104-out-of-band-state-sync.md

```diff
@@ -124,58 +138,173 @@ in order to adopt this solution.
 
 ## Decision
 
-> This section records the decision that was made.
-> It is best to record as much info as possible from the discussion that happened.
-> This aids in not having to go back to the Pull Request to get the needed information.
+As the SDK implemented a solution for all applications using it, and we do not have
+users requesting local state sync that are running in production, we will not implement the generally applicable
+solution. 
 
-State sync should support the bootstrap of new nodes from application snapshots
-obtained out-of-band by node operators.
+This ADR will outline both the solution implemented by the SDK, as well as a design proposal of a 
+generally applicable solution, that can be later on picked up and implemented in CometBFT.
 
-In other words, the following operation should be, in general terms, possible:
+## Detailed Design
 
-1. When configuring a new node, operators should be able to define where (e.g.,
-   a local file system location) the client side of state sync should look for
-   application snapshots to install or restore;
-1. Operators should be able to instruct the server side of state sync on a
-   running node to export application snapshots, when they are available,
-   to a given location (e.g., at the local file system);
-1. It is up to node operators to transfer application snapshots produced by the
-   running node (server side) to the new node (client side) to be bootstrapped
-   using this new state sync feature.
+### Application-implemented local state sync
 
+This section describes the solution to local state sync implemented by the SDK. An application can 
+chose to implement local state sync differently, or implement only a subset of the functionalities
+implemented by the SDK.
 
-## Detailed Design
+This solution exposes a command line interface enabling a node to manipulate the snapshots
+including dumping existing snapshots to an exportable format, loading and deleting exported snapshots,
+as well as bootstraping a node's state based on a local snapshot. 
+
+The SDK exposes the following commands for snapshot manipulation:
+
+```script
+
+delete      Delete a local snapshot
+dump        Dump the snapshot as portable archive format
+export      Export app state to snapshot store
+list        List local snapshots
+load        Load a snapshot archive file into snapshot store
+restore     Restore app state from local snapshot
+
+```
+
+and the following command to bootstrap the state of CometBFT upon a completed state sync:
+
+``` script
+
+comet bootstrap-state          Bootstrap the state of CometBFT's state and block store
+
+```
+
+These commands enable the implementation of both the client and server side of statesync. 
+Namely, a statesync server can use `dump` to create a portable archieve format out existing snapshots,
+or trigger snapshot creation using `export`. 
+
+The client side, restores the application state from a local snapshot that was previously exported, using
```

Comment:
The client should first `load` a snapshot from a file to then `restore` it. The latter restore/install a snapshot from the snapshot store, while the first adds a snapshot to the store.